miditok/pytorch_data/datasets.py

```diff
@@ -150,14 +162,24 @@ class DatasetTok(_DatasetABC):
     Note that if you directly load MIDI files, the loading can take some time as they will need to be tokenized.
     You might want to tokenize them before once with the `tokenizer.tokenize_midi_dataset()` method.
 
+    Additionally, you can use the `func_to_get_labels` argument to provide a method allowing to use labels (one label
+    per file).
+
     :param files_paths: list of paths to files to load.
     :param min_seq_len: minimum sequence length (in nb of tokens)
     :param max_seq_len: maximum sequence length (in nb of tokens)
     :param tokenizer: tokenizer object, to use to load MIDIs instead of tokens. (default: None)
     :param one_token_stream: give False if the token files contains multiple tracks, i.e. the first dimension of
         the value of the "ids" entry corresponds to several tracks. Otherwise, leave False. (default: True)
+    :param func_to_get_labels: a function to retrieve the label of a file. The method must take to positional
+            arguments: the first is either a MidiFile or the tokens loaded from the json file, the second is the path
+            to the file just loaded. The method must return an integer which correspond to the label id (and not the
+            absolute value, e.g. if you are classifying 10 musicians, return the id from 0 to 9 included corresponding
+            to the musician).
```

Comment:
Would it make sense to allow labels to be vectors instead of just ints? This would allow to have OHE labels for example.

Reply:
Yes indeed. Working on it!